circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -55,27 +55,28 @@ object JsonDebugErrorHandler {
                 s"""Message failure handling request: ${req.method} ${req.pathInfo} from ${req.remoteAddr
                     .getOrElse("<unknown>")}"""
               )
-              .unsafeRunSync()
-            val firstResp = mf.toHttpResponse[G](req.httpVersion)
-            Response[G](
-              status = firstResp.status,
-              httpVersion = firstResp.httpVersion,
-              headers = firstResp.headers.redactSensitive(redactWhen),
-            ).withEntity(JsonErrorHandlerResponse[G](req, mf)).pure[F]
+              .as {
+                val firstResp = mf.toHttpResponse[G](req.httpVersion)
+                Response[G](
+                  status = firstResp.status,
+                  httpVersion = firstResp.httpVersion,
+                  headers = firstResp.headers.redactSensitive(redactWhen),
+                ).withEntity(JsonErrorHandlerResponse[G](req, mf))
+              }
           case t =>
             serviceErrorLogger
```

Comment:
Currently, 'serviceErrorLogger' is defined as a `def`. Should we define it as a `val` right above the Kleisli? 

```scala
): Kleisli[F, Request[G], Response[G]] = {
  val serviceErrorLogger = LoggerFactory[F].getLoggerFromName(...)
  Kleisli { req =>
  ...
  }
}
```

Reply:
I don't see why not.

Reply:
Oh good point. We can probably do the same with `messageFailureLogger`?

Btw is there a meaningful difference between the effectful and non-effectful APIs for obtaining a logger?
https://github.com/typelevel/log4cats/blob/36bfbb6bd8d89e5f81575f85c78b0f8616de5c0b/core/shared/src/main/scala/org/typelevel/log4cats/LoggerFactoryGen.scala#L21-L26

Reply:
Is it worth making these loggers `lazy val`s?

Reply:
Done